Sources/Apollo/CacheWriteInterceptor.swift

```diff
@@ -69,6 +71,15 @@ public struct CacheWriteInterceptor: ApolloInterceptor {
                              request: request,
                              response: response,
                              completion: completion)
+
+      terminate(chain: chain, ifNeededFor: Operation.operationType)
+    }
+  }
+
+  private func terminate(chain: RequestChain, ifNeededFor operationType: GraphQLOperationType) {
+    // HTTP-based subscriptions are re-entrant so this may not be terminal if not a subscription
+    if operationType != .subscription {
```

Comment:
This is an example of the implementation detail that interceptor authors would be required to know for this solution; because this interceptor is after the `MultipartResponseParsingInterceptor` and HTTP-based subscriptions are re-entrant it means that `terminate` should only be called for non-subscription operations or in cases where it's known that a subscription event could never be received again.